### DISCUSSION TITLE
Azure App service

### DISCUSSION THREAD

**Discussion by jfmatth** [netbox-community/netbox#13043] [Ideas]
Has anyone tried to build this in Azure as an App Service?

Seems ideal.

I think ill give it a try and see what i can document.

**Comment by BrokenStandards** [comment]
Not sure if they are the same hing, but if you mean Azure AD Application / Enterprise Application then it is community documented with all types of connections be is SAML, OAuth2, maybe openid.  With SSO. Main issue is most of the doc are older from when things were harder, and official docs exclude mostly necessary custom role/group scripts for SSO.

Just barely finished setting up mine, but I am new to netbox and azure so was a bit much to learn. Using azure vm, keystore, postgres flexible server, vm hosted reddis since seemed expensive. ~~Working on getting media storage working but storage config doesn't seem to want AZURE_CONNECTION_STRING, or any of the keys defined... so still working on it.~~ Edit: All good now! Forgot to quate the key side of the dict/object/hash/whatever python calls that

**Reply by jfmatth** [reply]
Actually, I was referring to Azure App Service, their PAAS which will take the 'django' app and bundle it to run on their infra at whatever scale you like, then hookup the same stuff you did, Redis, Postgres, etc. on their infra.

Ill give it a try, looks pretty simple.

**Reply by craftzneko** [reply]
@jfmatth Hi am trying to set this up on azure app service now. Did you use docker compose option

**Reply by jfmatth** [reply]
I haven't put any time here, but was just going to see if the basic push worked, not use a Docker container to host.

**Reply by MFKDGAF** [reply]
Were you ever able to get this deployed on to Azure App Services?

I am trying to get this deployed on to Azure App Services but haven't been successful yet.

I've read the documentation but an am a bit confused as to what are the required settings that must be present in the docker compose file/App Services environment settings (app settings).

Also, it looks like when you build from source, it deploys 3 containers (Netbox,Postgres and redis). Is there not a single container image that includes all 3?
